src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -142,3 +112,120 @@ def get_disk_type(self):
+        event_files = [os.path.join(self.events_folder_path, event_file) for event_file in os.listdir(self.events_folder_path) if (event_file.lower().endswith(".json"))]
+        event_files.sort(key=os.path.getmtime, reverse=True)
+
+        for event_file in event_files:
+            try:
+                if self.__get_events_dir_size() < Constants.TELEMETRY_DIR_SIZE_LIMIT_IN_BYTES - Constants.TELEMETRY_EVENT_FILE_SIZE_LIMIT_IN_BYTES:
+                    # Not deleting any more event files as the event directory has sufficient space to add at least one new event file. Not printing this statement as it will add repetitive logs
+                    break
+
+                if os.path.exists(event_file):
+                    os.remove(event_file)
+                    self.composite_logger.log_telemetry_module("Deleted event file. [File={0}]".format(repr(event_file)))
+            except Exception as e:
+                self.composite_logger.log_telemetry_module_error("Error deleting event file. [File={0}] [Exception={1}]".format(repr(event_file), repr(e)))
+
+        if self.__get_events_dir_size() >= Constants.TELEMETRY_DIR_SIZE_LIMIT_IN_BYTES:
+            raise Exception("Older event files were not deleted. Current event will not be sent to telemetry as events directory size exceeds maximum limit")
+
+    def write_event_using_temp_file(self, folder_path, data, mode='w'):
+        """ Writes to a temp file in a single operation and then moves/overrides the original file with the temp """
+        file_path = self.__get_event_file_path(folder_path)
+        prev_events = []
+        try:
+            if os.path.exists(file_path):
+                file_size = self.get_file_size(file_path)
+                # if file_size exceeds max limit, sleep for 1 second, so the event can be written to a new file since the event file name is a timestamp
+                if file_size >= Constants.TELEMETRY_EVENT_FILE_SIZE_LIMIT_IN_BYTES:
+                    time.sleep(1)
+                    file_path = self.__get_event_file_path(folder_path)
+                else:
+                    prev_events = self.__fetch_events_from_previous_file(file_path)
+
+            prev_events.append(data)
+            with tempfile.NamedTemporaryFile(mode, dir=os.path.dirname(file_path), delete=False) as tf:
+                json.dump(prev_events, tf, default=data.__str__())
+                tempname = tf.name
+            shutil.move(tempname, file_path)
+        except Exception as error:
+            raise Exception("Unable to write to telemetry. [Event File={0}] [Error={1}].".format(str(file_path), repr(error)))
+
+    def __get_events_dir_size(self):
+        return sum([os.path.getsize(os.path.join(self.events_folder_path, f)) for f in os.listdir(self.events_folder_path) if os.path.isfile(os.path.join(self.events_folder_path, f))])
```

Comment:
please break this line apart into multiple lines of execution -- doesn't lend to long term readability/maintainability as it is. Clarity > consiseness.